File: server-side/Models/Task.cs
using System.ComponentModel.DataAnnotations;

namespace TaskMonitor.Models
{
    public class Task
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();
        public Guid ProjectId { get; set; }

        [MaxLength(250)]
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; }
        public DateTime? UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }

        public virtual ICollection<TimeTracker> TimeTrackers { get; } = [];

        public Project? Project { get; set; }
    }
}

File: client-side/src/typings/models/task.ts
import type { Project } from './project';
import type { TimeTracker } from './timeTracker';

export interface Task {
  id: string;
  projectId: string;
  name: string;
  description: string;
  createdAt: string | number;
  updatedAt?: string | number;
  deletedAt?: string | number;

  timeTrackers: TimeTracker[];

  project?: Project;
}
